app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -12,19 +17,18 @@ open class FeedViewModel : ViewModel() {
     private val stateInternal: MutableLiveData<State> = MutableLiveData<State>(DEFAULT_STATE)
     private val networkErrorEvent = MutableLiveData<Event<String>>()
 
+    private lateinit var apiServiceDisposable: Disposable
```

Comment:
lateinit is in general bad and we would avoid using them, lateinit means the property is initialized later and not during the object creation. And that exactly why it is bad because you don't know when it is about to be initialized. So replace it by a nullable disposable or by an empty disposable that does nothing. ```apiServiceDisposable = Disposables.empty()```

Reply:
I would also rename the property as `refreshTaskDisposable` or something like that.